src/vulkan/vulkan_renderer.h

```diff
@@ -11,27 +11,28 @@
 namespace IC {
     class VulkanRenderer : public Renderer {
     public:
-        VulkanRenderer(RendererConfig &config);
-        ~VulkanRenderer();
+        VulkanRenderer(const RendererConfig &config);
+        virtual ~VulkanRenderer();
```

Comment:
what does virtual destructor do?

Reply:
Apparently it's undefined behavior to not have it here: [link](https://en.cppreference.com/w/cpp/language/virtual#Virtual_destructor)
"...if the destructor of the base class is not virtual, deleting a derived class object through a pointer to the base class is undefined behavior"
Basically you have to specifiy a virtual dtor whenever you want a non-default dtor for a virtual class. Otherwise it'll usually not call the whole inheritence chain of dtor's, but it's technically undefined behavior so who knows what would happen

Reply:
Technically in this case it's not required because it's already specified as virtual by the base class. I guess I just added it for verbosity 🤷